lms/djangoapps/ccx/tests/test_views.py

```
@@ -196,6 +197,71 @@ def test_create_ccx(self):
+            sequentials = flatten(
+                [chapter.get('children', []) for chapter in chapters])
+            verticals = flatten(
+                [sequential.get('children', []) for sequential in sequentials])
+            self.assertEqual(n_chapters, len(chapters))
+            self.assertEqual(n_sequentials, len(sequentials))
+            self.assertEqual(n_verticals, len(verticals))
+            all_elements = chapters + sequentials + verticals
+            return [elem['location'] for elem in
+                    all_elements if 'location' in elem]
+
+
+        today.return_value = datetime.datetime(2014, 11, 25, tzinfo=pytz.UTC)
+        self.make_coach()
+        ccx = self.make_ccx()
+        url = reverse(
+            'ccx_coach_dashboard',
+            kwargs={
+                'course_id': CCXLocator.from_course_locator(
+                    self.course.id, ccx.id)
+            }
+        )
+        mstore = modulestore()
+        # all the elements are visible
+        assert_elements_in_schedule(url)
+        vertical = self.verticals[0]
+        vertical.visible_to_staff_only = True
+        mstore.update_item(vertical, self.coach.id)
+        locations = assert_elements_in_schedule(url, n_verticals=7)
+        self.assertNotIn(unicode(vertical.location), locations)
+        # hide a sequential
+        sequential = self.sequentials[0]
+        sequential.visible_to_staff_only = True
+        mstore.update_item(sequential, self.coach.id)
+        locations = assert_elements_in_schedule(
+            url, n_sequentials=3, n_verticals=6)
+        self.assertNotIn(unicode(sequential.location), locations)
+        # hide a chapter
+        chapter = self.chapters[0]
+        chapter.visible_to_staff_only = True
+        mstore.update_item(chapter, self.coach.id)
+        locations = assert_elements_in_schedule(
+            url, n_chapters=1, n_sequentials=2, n_verticals=4)
+        self.assertNotIn(unicode(chapter.location), locations)
```

Well my worry with having a bunch of asserts in one test, that are testing different conditions, is that failures can be hidden by other earlier failures. I like to try to keep unit tests scoped as small as possible. If this must all be in one test, so be it.